### PR TITLE
Adding tests for redirects/noindex

### LINKED ISSUE #127: Create tests for template logic
Work is currently being done to redirect pages and apply "nofollow" attributes to URLs via meta tags. The logic for this lives in liquid templates.

A small test suite that checked combinations of environment variables and URLs for the presence or absence of these meta tags would help us check that our logic is correct, reduce reliance on manual testing, and serve as guardrails for future changes as we add more guides.

If pursuing this, the task would likely need to be broken up into smaller issues.  For example: identifying testing the library we want to use for testing, hooking it up in 11ty and adding it to CI, and creating tests for specific cases.

### PR DESCRIPTION
**Pull request by eric-gade**
## Changes proposed in this pull request:
- Mocha and chai dependencies
- Tests for redirects and noindex meta tags across different configs (env variables etc)
- workflow updates to run tests in CI

## Instructions
To run tests locally:
```sh
npm run test:redirects
```

## security considerations
No known concerns
  
## Closes
#127

### CODE REVIEW

File: tests/meta-tag-tests.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
+/* eslint no-unused-expressions: "off" */
```
Comment (jduss4) [inline tests/meta-tag-tests.js:1]: Question: Why is this detection turned off?

File: tests/meta-tag-tests.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+const chai = require('chai');
+const yaml = require('js-yaml');
+const fs = require('fs');
+const path = require('path');
+const childProcess = require('child_process');
+const jsdom = require('jsdom');
+
+const { JSDOM } = jsdom;
+const {expect} = chai;
```
Comment (jduss4) [inline tests/meta-tag-tests.js:10]: Question or nitpick:  I'm a JS novice, does the space matter for this syntax?  Is there a reason one has spaces and the other doesn't?

File: tests/meta-tag-tests.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  .map((dirEntry) => dirEntry.name);
+const REDIRECTED_GUIDE_NAMES = Object.keys(redirects);
+const REPLATFORMED_GUIDE_NAMES = GUIDE_NAMES.filter(
+  (name) => !REDIRECTED_GUIDE_NAMES.includes(name),
+);
+
+const runEleventy = async (envName = '') => {
+  childProcess.execSync('npx @11ty/eleventy', {
+    cwd: path.resolve(__dirname, '..'),
+    timeout: 15000,
```
Comment (jduss4) [inline tests/meta-tag-tests.js:27]: Suggestion: Should this timeout be set using the same value you are passing into the mocha command in `package.json` ?

**Review by jduss4 — COMMENTED** [review]
I had a couple questions but the test cases look good and it's very clear what they are checking and how!

**Comment by jduss4** [comment]
Does the guides team have a preferred system for commit history?  I was wondering if it makes sense to squash quite a few of these down, as this PR is a pretty self-contained thing and probably doesn't need a ton of explanation in terms of individual commits.

**Comment by echappen** [comment]
@jduss4 I don't believe we have a shared git strategy yet—it's something we plan to put in our [team doc](https://docs.google.com/document/d/1iBmKToTK7js65aOsljXirK0yLh--QZ_h6ugXtvqbuRg/edit#heading=h.xfl9ebdv8hs4). For now I'd say do what makes most sense to you.

File: tests/meta-tag-tests.js (hunk 3 above)
Comment (eric-gade) [reply tests/meta-tag-tests.js]: Good call, addressed here: https://github.com/18F/guides/pull/132/commits/8f4f99d37868d6712623a5d71f9ff1af9fd4232d

File: tests/meta-tag-tests.js (hunk 1 above)
Comment (jduss4) [reply tests/meta-tag-tests.js:1]: Oh gotcha!

**Review by jduss4 — APPROVED** [review]

**Pull request merged**